MCO-1316 On-Cluster Layering GA - 4.18 upgrades and integrations (#87730)
openshift/openshift-docs: modules/coreos-layering-configuring-on-revert.adoc

// Module included in the following assemblies:
//
// * machine_configuration/coreos-layering.adoc

:_mod-docs-content-type: PROCEDURE
[id="coreos-layering-configuring-on-revert_{context}"]
= Reverting an on-cluster custom layered image

You can revert an on-cluster custom layered image from nodes by removing the label for the machine config pool (MCP) that you specified in the `MachineOSConfig` object. After you remove the label, the Machine Config Operator (MCO) reboots the nodes in that MCP with the cluster base {op-system-first} image, along with any previously-made machine config changes, overriding the custom layered image. 

[IMPORTANT]
====
If the node where the custom layered image is deployed uses a custom machine config pool, before you remove the label, make sure the node is associated with a second MCP.
====

You can reapply the custom layered image to the node by using the `oc label node/<node_name> 'node-role.kubernetes.io/<mcp_name>='` label.

.Prerequisites

* You have opted-in to on-cluster layering by creating a `MachineOSConfig` object.

.Procedure

> * Remove the remove the label from the node by using the following command:

Reviewer: ```suggestion
* Remove the label from the node by using the following command:
```

> * Remove the remove the label from the node by using the following command:

Reviewer 2: ```suggestion
* Remove the label from the node by using the following command:
```

+
[source,terminal]
----
$ oc label node/<node_name> node-role.kubernetes.io/<mcp_name>-
----
+
When you save the changes, the MCO drains, cordons, and reboots the nodes. After the reboot, the node uses the cluster base {op-system-first} image.

.Verification

> You can verify that the custom layered image is removed by performing any of the following checks:

Reviewer 3: This sounds like you only need to do one, is that the case? If so, I'd make the below items unnumbered bullets.

Or if they aren't, then I'd remove "any" here

```suggestion
You can verify that the custom layered image is removed by performing the following checks:
```

. Check the nodes to see that scheduling on the nodes is disabled. This indicates that the change is being applied:
+
[source,terminal]
----
$ oc get nodes
----
+
.Example output
[source,terminal]
----
NAME                                         STATUS                     ROLES                  AGE   VERSION
ip-10-0-148-79.us-west-1.compute.internal    Ready                      worker                 32m   v1.31.3
ip-10-0-155-125.us-west-1.compute.internal   Ready,SchedulingDisabled   worker                 35m   v1.31.3
ip-10-0-170-47.us-west-1.compute.internal    Ready                      control-plane,master   42m   v1.31.3
ip-10-0-174-77.us-west-1.compute.internal    Ready                      control-plane,master   42m   v1.31.3
ip-10-0-211-49.us-west-1.compute.internal    Ready                      control-plane,master   42m   v1.31.3
ip-10-0-218-151.us-west-1.compute.internal   Ready                      worker                 31m   v1.31.3
----

. When the node is back in the `Ready` state, check that the node is using the base image:

.. Open an `oc debug` session to the node. For example:
+
[source,terminal]
----

> $ oc debug node/ip-10-0-155-125.us-west-1.compute.internal

Reviewer: ```suggestion
$ oc debug node/<node_name>
```

----

.. Set `/host` as the root directory within the debug shell:
+
[source,terminal]
----
sh-5.1# chroot /host
----

> .. Run an `rpm-ostree status` command to view that the custom layered image is in use:

Reviewer 4: ```suggestion
.. Run an `rpm-ostree status` command to view that the base image is in use:
```

+
[source,terminal]
----
sh-5.1# rpm-ostree status
----
+
.Example output
+

> ----

Reviewer 3: This needs a language
